src/validation.cpp

```diff
@@ -3860,6 +3939,9 @@ bool ChainstateManager::ProcessNewBlock(const CChainParams& chainparams, const s
         // Therefore, the following critical section must include the CheckBlock() call as well.
         LOCK(cs_main);
 
+        // Before doing any checks, begin warming the cache to process this block
+        ::ChainstateActive().WarmBlock(pblock);
```

Comment:
Are we always guaranteed to have checked the pow by this point? I *think* so, because of AcceptBlockHeader?

Worth documenting that this is expected or putting the WarmBlock call inside CheckBlock after the PoW check.

Reply:
Updated to explicitly check PoW here if we haven't already.